core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -861,6 +862,19 @@ public static Builder buildFromEmpty() {
     return new Builder();
   }
 
+  public static Builder buildFromMetadataUpdates(List<MetadataUpdate> metadataUpdates) {
```

Comment:
I don't think we should expose this in `TableMetadata`. It should be fine to have this inside `CatalogHandlers#create`

Reply:
Thanks for your suggestion, that seems better, fixed!